File: fortran_pFUnit/src/speed_calc.f90
! Simple program to return the speed of sound
program speed_calc

  use set_precision,   only : dp
  use fluid_constants, only : gamma, set_fluid_constants

  real(dp) :: rho, p, a

  interface
    function speed_of_sound(p,rho)
      use set_precision, only : dp
      real(dp), intent(in) :: p, rho
      real(dp)             :: speed_of_sound
    end function speed_of_sound
  end interface

  continue

  ! Set up fluid constants
  gamma = 1.4_dp
  call set_fluid_constants

  print*, "Enter the pressure:"
  read(*,*) p

  print*, "Enter the density:"
  read(*,*) rho

  print*, speed_of_sound(p,rho)

end program speed_calc

include 'speed_of_sound.f90'

File: cxx_catch2/src/speed_calc.cpp
#include <iostream>
#include "functions/speed_of_sound.h"

// Simple program to return the speed of sound
int main(int argc, char* argv[]) {
    // Set up fluid constants
    double gamma = 1.4;

    double rho, p;

    std::cout << "Enter the pressure: ";
    std::cin >> p;

    std::cout << "Enter the density: ";
    std::cin >> rho;

    std::cout << "Speed of sound: " << speed_of_sound(p, rho, gamma) << std::endl;
}
